plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/tasks/GenerateReportsTask.kt

```diff
@@ -90,6 +90,8 @@ abstract class GenerateReportsTask @Inject constructor(
     @get:Optional
     internal abstract val baseline: RegularFileProperty
 
+    private val rootDir: File = project.rootDir
```

Comment:
Does this need to be class scoped? Or can it be scoped smaller, to only the `init` block? Also, can you elaborate on why this fixes this, or, at minimum, can you include an explicit comment in the code explaining why this is extracted to a variable eagerly. That way the rationale for this change isn't lost in the future? 🙂

Reply:
According to https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:requirements:~:text=Use%3A-,project.rootDir,-A%20task%20input for configuration caching rootDir you should use
`A task input or output property or a script variable to capture the result of using project.rootDir to calculate the actual parameter.`
So according to https://docs.gradle.org/current/userguide/configuration_cache.html#config_cache:requirements:use_project_during_execution `A task must not use any Project objects at execution time. This includes calling Task.getProject() while the task is running.` and to achieve that I was calling it within the class. I could also let the input set the relative rootdir path, but I thought this is easier to grasp? I added a comment to hopefully make it understandable.

What do you mean with init block scoped? There is no difference to this and 
```
private val rootDir: File
init {
     rootDir = project.rootDir
}
``
`
or do I misunderstand you?

Reply:
Sorry, I misunderstood the code. I thought the `rootDir` was only used inside of the `init` block, but re-reading the code, I was completely wrong. My mistake.

Thanks for adding a comment inline in the code! 🙂